### ISSUE TITLE
False positive in Redundant Set Access Control Rule

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#2182]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version (master)
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Bug Report

Private has different meanings when at the top-level vs nested. Top-level effectively means "fileprivate" whereas nested means "private within this scope".

Repro:
```swift
private final class A {
    private(set) var value: Int
    ^ warning: Redundant Set Access Control Rule Violation: Property setter access level shouldn't be explicit if it's the same as the variable access
}
```

In this case, the code is acceptable and we shouldn't require `fileprivate final class A`.

This rule was added in #2036.
